test/models/PaymentAllocationsResponseModelSpec.scala

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright 2020 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package models
+
+import assets.BaseTestConstants._
+import assets.PaymentAllocationsTestConstants._
+import models.paymentAllocations._
+import org.scalatest.Matchers
+import play.api.libs.json.{JsSuccess, Json}
+import uk.gov.hmrc.play.test.UnitSpec
+
+class PaymentAllocationsResponseModelSpec  extends UnitSpec with Matchers {
+
+  "The PaymentAllocationsModel" should {
+    "have the amount as '110.10'" in {
+      testValidPaymentAllocationsModel.amount.get shouldBe 110.10
+    }
+
+    "have the method as 'Payment by Card'" in {
+      testValidPaymentAllocationsModel.method.get shouldBe "Payment by Card"
+    }
+
+    "have the transaction date as 27th May 2019" in {
+      testValidPaymentAllocationsModel.transactionDate.get shouldBe "2019-05-27"
+    }
+
+    "for the 1st AllocationDetail" should {
+
+      val allocation = testValidPaymentAllocationsModel.allocations.head
+
+      "have the transaction id as '1040000872'" in {
+        allocation.transactionId.get shouldBe "1040000872"
+      }
+
+      "have the from date as 27th June 2019" in {
+        allocation.from.get shouldBe "2019-06-27"
+      }
+
+      "have the to date as 27th August 2019" in {
+        allocation.to.get shouldBe "2019-08-27"
+      }
+
+      "have the type as '1481'" in {
+        allocation.`type`.get shouldBe "1481"
+      }
+
+      "have the amount as '10.10'" in {
+        allocation.amount.get shouldBe 10.10
+      }
+
+      "have the clearedAmount as '5.50'" in {
+        allocation.clearedAmount.get shouldBe 5.50
+      }
+    }
+
+    "for the 2nd AllocationDetail" should {
+
+      val allocation = testValidPaymentAllocationsModel.allocations.last
+
+      "have the transaction id" in {
+        allocation.transactionId.get shouldBe "1040000873"
+      }
+
+      "have the from date as 28th July 2019" in {
+        allocation.from.get shouldBe "2019-07-28"
+      }
+
+      "have the to date as 28th September 2019" in {
+        allocation.to.get shouldBe "2019-09-28"
+      }
+
+      "have the type as '1482'" in {
+        allocation.`type`.get shouldBe "1482"
+      }
+
+      "have the amount as '10.10'" in {
+        allocation.amount.get shouldBe 10.90
+      }
+
+      "have the clearedAmount as '5.50'" in {
+        allocation.clearedAmount.get shouldBe 5.90
+      }
+    }
```

Comment:
I don't think we should be testing this, at this point you are just testing the functionality of scala

Reply:
Yeah, even I was thinking same but scoverage plugin reports this as not covered in coverage if not tested.
I just noticed that we ignore models package from scoverage, so removed these tests now :)